proto-lens-protoc/src/Data/ProtoLens/Compiler/Combinators.hs

```diff
@@ -161,16 +174,21 @@ match n ps e = Syntax.Match () n ps (Syntax.UnGuardedRhs () e) Nothing
 
 type Module = Syntax.Module ()
 
-module' :: ModuleName -> [ModulePragma] -> [Syntax.ImportDecl ()] -> [Decl] -> Module
-module' modName
+type ExportSpec = Syntax.ExportSpec ()
+
+module'
+    :: ModuleName
+    -> Maybe [ExportSpec]
+    -> [ModulePragma]
+    -> [Syntax.ImportDecl ()]
+    -> [Decl]
+    -> Module
+module' modName exports
     = Syntax.Module ()
         (Just $ Syntax.ModuleHead () modName
                     -- no warning text
                     Nothing
-                    -- no explicit exports; we export everything.
-                    -- TODO: Also export public imports, taking care not to
-                    -- cause a name conflict between field accessors.
-                    Nothing)
+                    (Syntax.ExportSpecList () <$> exports))
```

Comment:
what's the reason for this change?

Reply:
It was part of #136 (now merged).  Sorry for the noise.

Previously the modules that we generated always exported everything via `module M where`, so the export list parameter was always `Nothing`.  Now, for the types module we need to pass an explicit list, i.e., `module M (...) where` instead, so it can export pattern synonyms in a nicer way.